Src/Web/Web.Shared.Net/ApplicationInsightsHttpModule.cs

```diff
@@ -81,6 +111,17 @@ private void OnEndRequest(object sender, EventArgs eventArgs)
 
                 if (this.IsFirstRequest(httpApplication))
```

Comment:
Something feels weird to me about "is first" logic but I don't currently recall why...